### PR TITLE
chore: update @octokit/rest from v16 to v17

### PR DESCRIPTION
**Pull request by Trott**
Replace deprecated .hasLastPage() with pagination. This enables us to bump @octokit/rest from v16 to v17.

### CODE REVIEW

**Comment by Trott** [comment]
@nodejs/github-bot

File: lib/push-jenkins-update.js [hunk 1]
```diff
@@ -77,20 +77,16 @@ function findPrInRef (gitRef) {
   return parseInt(gitRef.split('/')[2], 10)
 }
 
-async function findLatestCommitInPr (options, pageNumber = 1) {
-  const res = await githubClient.pulls.listCommits({
+async function findLatestCommitInPr (options) {
+  const { Octokit } = require('@octokit/rest')
+  const githubClient = new Octokit({})
```
Comment (phillipj) [inline lib/push-jenkins-update.js:82]: Was it deliberate to not use `githubClient` from line 3?

File: test/unit/push-jenkins-update.test.js [hunk 2]
```diff
@@ -4,32 +4,41 @@ const { findLatestCommitInPr } = require('../../lib/push-jenkins-update')
...
 const readFixture = require('../read-fixture')
-const { ignoreQueryParams } = require('../common')
 
 tap.test('findLatestCommitInPr: paginates results when more than 100 commits in a PR', async (t) => {
-  const commitsFixturePage1 = readFixture('pull-requests-commits-page-1.json')
-  const commitsFixturePage104 = readFixture('pull-requests-commits-page-104.json')
+  const commitsFixturePage1 = readFixture('pull-request-commits-page-1.json')
+  const commitsFixturePage2 = readFixture('pull-request-commits-page-2.json')
+  const commitsFixturePage3 = readFixture('pull-request-commits-page-3.json')
+  const commitsFixturePage4 = readFixture('pull-request-commits-page-4.json')
```
Comment (phillipj) [inline test/unit/push-jenkins-update.test.js:12]: Hooray for tests being maintained and adjusted accordingly

**Comment by phillipj** [comment]
@Trott sorry for the late reply, hectic week on my end...

Looks promising from a distance 👍🏻

File: lib/push-jenkins-update.js (hunk 1 above)
Comment (Trott) [reply lib/push-jenkins-update.js:82]: That was probably a blunder. Sorry about that!

**Comment by phillipj** [comment]
No worries, as long as it works. I was mostly asking in case it was a
workaround for whatever reason.

File: lib/push-jenkins-update.js (hunk 1 above)
Comment (richardlau) [reply lib/push-jenkins-update.js:82]: Is this why we're hitting rate limits? The github-bot logs now have lots of "API rate limit exceeded for 23.253.100.79. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)" errors, e.g.
```console
{"name":"bot","hostname":"infra-rackspace-debian8-x64-1","pid":15899,"req_id":"e62c49d0-bbaa-11ed-9410-77b81a7c3334","identifier":"node-test-linux-ubuntu1804-64","event":"end","level":50,"err":{"message":"API rate limit exceeded for 23.253.100.79. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","name":"HttpError","stack":"HttpError: API rate limit exceeded for 23.253.100.79. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)\n    at /home/iojs/github-bot/node_modules/@octokit/request/dist-node/index.js:86:21\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at async Object.next (/home/iojs/github-bot/node_modules/@octokit/plugin-paginate-rest/dist-node/index.js:112:28)\n    at async findLatestCommitInPr (/home/iojs/github-bot/lib/push-jenkins-update.js:88:23)","code":403},"msg":"Error while handling Jenkins end event","time":"2023-03-05T23:10:20.025Z","v":0}
```

Comment (Trott) [reply lib/push-jenkins-update.js:82]: Ugh, sending a PR momentarily. I wouldn't have put these things together. Sorry about that.

Comment (phillipj) [reply lib/push-jenkins-update.js:82]: Thanks for opportunity to learn why Github requests better be authenticated 👍🏻

**Pull request merged**